### PR TITLE
[#56] test script for refund transaction

### PR DESCRIPTION
**Pull request by trojkat**
```
╰(clove)─λ bin/test-refund-transaction.py -tx bcbcab25569c2104e51746fdda4158aee4735d19221aa7c946079a0b89430f1a -p cRoFBWMvcLXrLsYFt794NRBEPUgMLf5AmnJ7VQwiEenc34z7zSpK
>>> Searching for transaction: bcbcab25569c2104e51746fdda4158aee4735d19221aa7c946079a0b89430f1a
>>> Contract details:
{'locktime': datetime.datetime(2018, 2, 14, 15, 4, 39),
 'recipient_address': 'msJ2ucZ2NDhpVzsiNE5mGUFzqFDggjBVTM',
 'refund_address': 'mmJtKA92Mxqfi3XdyGReza69GjhkwAcBN1',
 'secret_hash': '7bb0eaadd75101d5c3eef24b743568f177c228768175ff4ccc44d5de607dad44',
 'transaction_hash': 'bcbcab25569c2104e51746fdda4158aee4735d19221aa7c946079a0b89430f1a',
 'value': 0.001,
 'value_text': '0.00100000 BTC'}
>>> Refund transaction ready to be published
{'fee': 0.00054743,
 'fee_per_kb': 0.00243303,
 'fee_per_kb_text': '0.00243303 BTC / 1 kB',
 'fee_text': '0.00054743 BTC',
 'recipient_address': 'mmJtKA92Mxqfi3XdyGReza69GjhkwAcBN1',
 'size': 226,
 'size_text': '226 bytes',
 'transaction': '01000000011a0f43890b9a0746c9a71a22195d73e4ae5841dafd4617e504219c5625abbcbc000000006c483045022100d41ef1cd00d45bccc2f1069c8dbb9ea2f5430f958faa6800b88e6e60c7ebb1b60220308826b21cfba517560e2c835c7e38ea0eb09d16c2128a9746616538b65a6f5f012102187b57bba6e143aca9da3557fe8bcf912379ca7ffb8c1967ca6dbde2dc695f19000000000001c9b00000000000001976a9143f8870a5633e4fdac612fba47525fef082bbe96188acf741845a',
 'transaction_hash': 'c0d8c9c49e1f0d1d481f86a6088f0cf392897e4db477fdc4548d5ce98f3307b2',
 'value': 0.001,
 'value_text': '0.00100000 BTC'}
Do you want to publish this transaction (y/n): y
>>> Publishing transaction
>>> Transaction published!
>>> https://live.blockcypher.com/btc-testnet/tx/c0d8c9c49e1f0d1d481f86a6088f0cf392897e4db477fdc4548d5ce98f3307b2/
```

### CODE REVIEW

**Review by pafff — APPROVED** [review]

File: bin/test-refund-transaction.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    pprint(details)
+
+    publish = input('Do you want to publish this transaction (y/n): ')
+    if publish != 'y':
+        exit('>>> Bye!')
+
+    print('>>> Publishing transaction')
+    refund_transaction.publish()
+
+    print('>>> Transaction published!')
```
Comment (szerepak) [inline bin/test-refund-transaction.py:45]: Please add Fore.GREEN here and 🎉 (tada sign - utf8 code: U+1F389)

**Review by szerepak — APPROVED** [review]
Just two _prettify_ things ;)

File: bin/test-refund-transaction.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    refund_transaction = contract.refund(wallet)
+    refund_transaction.add_fee_and_sign()
+
+    print('>>> Refund transaction ready to be published')
+    details = refund_transaction.show_details()
+    pprint(details)
+
+    publish = input('Do you want to publish this transaction (y/n): ')
+    if publish != 'y':
+        exit('>>> Bye!')
```
Comment (szerepak) [inline bin/test-refund-transaction.py:40]: https://www.fileformat.info/info/unicode/char/1f44b/index.htm (Fore.BLUE)

File: bin/test-refund-transaction.py (hunk 1 above)
Comment (trojkat) [reply bin/test-refund-transaction.py:45]: Ok, I will add colors for all of the scripts in next PR.

**Pull request merged**